File: src/main/java/com/teamwizardry/librarianlib/client/fx/shader/LibShaders.java
package com.teamwizardry.librarianlib.client.fx.shader;

import com.teamwizardry.librarianlib.client.fx.shader.uniforms.FloatTypes;

public enum LibShaders {
	INSTANCE;
	
	public static Shader HUE;
	
	private LibShaders() {
		initShaders();
	}
	
	private void initShaders() {
		HUE = new HueShader(null, "/assets/librarianlib/shaders/hue.frag");
		ShaderHelper.addShader(HUE);
	}
	
	public static class HueShader extends Shader {

		public HueShader(String vert, String frag) {
			super(vert, frag);
		}

		public FloatTypes.Float hue;
		
		@Override
		public void initUniforms() {
			hue = getUniform("hue");
		}

		@Override
		public void uniformDefaults() {
			hue.set(0);
		}
		
	}
}

File: src/main/java/com/teamwizardry/librarianlib/client/fx/shader/LibShaders.kt
package com.teamwizardry.librarianlib.client.fx.shader

import com.teamwizardry.librarianlib.client.fx.shader.uniforms.FloatTypes
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly

@SideOnly(Side.CLIENT)
object LibShaders {
    @JvmField
    val HUE = HueShader(null, "shaders/hue.frag")

    init {
        ShaderHelper.addShader(HUE)
    }

    class HueShader(vert: String?, frag: String?) : Shader(vert, frag) {

        var hue: FloatTypes.FloatUniform? = null

        override fun initUniforms() {
            hue = getUniform<FloatTypes.FloatUniform>("hue")
        }

        override fun uniformDefaults() {
            hue?.set(0f)
        }

    }
}
